met-api/src/met_api/models/map.py

```diff
@@ -0,0 +1,59 @@
+"""WidgetMap model class.
+
+Manages the map widget
+"""
+from __future__ import annotations
+from datetime import datetime
+from sqlalchemy.dialects.postgresql import JSON
+from sqlalchemy.sql.schema import ForeignKey
+from .base_model import BaseModel
+from .db import db
+from .default_method_result import DefaultMethodResult
+
+
+class WidgetMap(BaseModel):  # pylint: disable=too-few-public-methods
+    """Definition of the Map entity."""
+
+    __tablename__ = 'map'
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    widget_id = db.Column(db.Integer, ForeignKey('widget.id', ondelete='CASCADE'), nullable=True)
+    engagement_id = db.Column(db.Integer, ForeignKey('engagement.id', ondelete='CASCADE'), nullable=True)
+    title = db.Column(db.String(50))
+    latitutde = db.Column(db.Float)
```

Comment:
lat and long should be nullable=False I think

Reply:
what do we display if we have a map without longitude and latitude is that if it's just a shapefile

Reply:
We are not uploading shapefiles now, so let's not worry about that. And we shouldn't at the moment create these widgets without lats and longs